src/lib/upload-to-s3.js

```diff
@@ -3,22 +3,48 @@ const AWS = require('aws-sdk')
 const bucket = 'ccshutterbug'
 const s3 = new AWS.S3()
 
-module.exports = function uploadToS3 (key, buffer, contentType) {
-  return new Promise((resolve, reject) => {
-    console.log('uploading', key, 'to S3 bucket...')
-    console.time('upload to s3')
-    s3.upload({
-      Bucket: bucket,
-      Key: key,
-      ContentType: contentType,
-      Body: buffer
-    }, function (err, data) {
-      if (err) {
-        reject(err)
-      } else {
-        resolve(data.Location)
-      }
-      console.timeEnd('upload to s3')
-    })
-  })
+// This might see overcomplicated, but  S3 upload seems to randomly get stack once in a while.
+// See: https://www.pivotaltracker.com/story/show/183543485
+const MAX_ATTEMPTS = 5
+const BASE_TIMEOUT = 5000 // usually S3 upload takes around ~100ms, so that's plenty of time
```

Comment:
Unless you've adjusted it, I think the lambda timeout is only 30 seconds so 5+10+15+20+25 will be longer than that.

Reply:
The timeout used to be set to 90 seconds. But that seemed way too long; the longest successful requests took around 15 seconds. So, recently I changed the setting to 30 seconds indeed. 

Personally, I wouldn't worry about this code matching the AWS console settings. We might change these settings and forget to update the code or another way. And there are other time-consuming actions that are hard to predict. Realistically, if the upload fails 3 times, we can probably give up at this point too.

I've updated the base timeout to 3.5s, as that's still a lot, and if something gets stuck, the second attempt will happen faster. I've also changed the equation to BASE_TIMEOUT * Math.sqrt(attempt). That theoretically fits into 30s: 3.5 * (1 + Math.sqrt(2) + Math.sqrt(3) + Math.sqrt(4) + Math.sqrt(5)) = 29.33816321604617.